### PR TITLE
Signin with google

### LINKED ISSUE #59: logout
### Function
Write a function to logout from the app and when the user clicks on the logout button will be not authorized to see the authentication pages.

### LINKED ISSUE #62: sign in with google | firebase
- Create sign-in component and  import the following:
  (useContext, FirebaseAuth, AuthContext, useContext).
- Get the user state from the context.
- Implement config for FirebaseAuth for GoogleAuthProvider.
- If user exists or signed in, we redirect the page to home, else display the sign in methods with FirebaseAuth.

### PR DESCRIPTION
**Pull request by Israa91**
Relates #62 #59
### loginWithGoogle file
- Create Google auth provider and set its properties
- In the callback take the auth result and pass it as parameter to addUser function which adds the user to firestore.

### context.js file
- In useEffect add onAuthStateChange which returns a null value if the user logged out, and auth results if user logged in
- Set the value of authResults in user

### Header
- Import user context and use it to change header values

### CODE REVIEW

File: src/components/Layout/Header/index.js [hunk 1]
```diff
@@ -90,38 +99,35 @@ const Header = ({ isLogged, userImage, userName }) => {
...
               className="userImage"
             />
-            <Text className="usernameTitle">{userName}</Text>
+            <Text className="usernameTitle">{user.name}</Text>
             <MainButton
               icon={<LogoutOutlined />}
               buttName="logout"
               id="logout"
               className="logout"
+              onClick={() => app.auth().signOut()}
```
Comment (lina-jamal) [inline src/components/Layout/Header/index.js]: when the user log out 
the state user will be null
so you can make
setUser(null)

File: src/Login/loginWithGoogle.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    prompt: 'select_account',
+  });
+  firebase
+    .auth()
+    .signInWithPopup(provider)
+    .then((authResult) => {
+      const { user } = authResult;
+      addUser(user).then((res) => res);
+    })
+    .catch((error) => error);
```
Comment (lina-jamal) [inline src/Login/loginWithGoogle.js]: if the error happen you can set it in error state

File: src/Login/loginWithGoogle.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+const loginWithGoogle = () => {
+  const provider = new firebase.auth.GoogleAuthProvider();
+  provider.setCustomParameters({
+    prompt: 'select_account',
+  });
+  firebase
+    .auth()
+    .signInWithPopup(provider)
+    .then((authResult) => {
```
Comment (lina-jamal) [inline src/Login/loginWithGoogle.js]: please use async-await instead of promise

File: src/firebase/context.js [hunk 4]
```diff
@@ -5,13 +5,23 @@ import firebase from './config';
...
-  const [isSignedIn, setIsSignedIn] = useState(false);
+  const [user, setUser] = useState(null);
   const [error, setError] = useState(null);
 
   useEffect(() => {
     try {
-      firebase.auth().onAuthStateChanged(setIsSignedIn);
+      firebase.auth().onAuthStateChanged((userAuth) => {
+        if (userAuth) {
+          setUser({
```
Comment (lina-jamal) [inline src/firebase/context.js]: we need user id also

**Review by Sondos-Alalami — DISMISSED** [review]

**Review by alaa-sahloub — APPROVED** [review]

**Review by Sondos-Alalami — APPROVED** [review]

**Review by lina-jamal — APPROVED** [review]

**Pull request merged**